File: parser/src/main/java/io/ticofab/androidgpxparser/parser/domain/Bounds.java
package io.ticofab.androidgpxparser.parser.domain;

public class Bounds {

    private final Double mMinLat;
    private final Double mMinLon;
    private final Double mMaxLat;
    private final Double mMaxLon;

    private Bounds(Builder builder) {
        mMinLat = builder.getMinLat();
        mMinLon = builder.getMinLon();
        mMaxLat = builder.getMaxLat();
        mMaxLon = builder.getMaxLon();
    }

    public Double getMinLat() {
        return mMinLat;
    }

    public Double getMinLon() {
        return mMinLon;
    }

    public Double getMaxLat() {
        return mMaxLat;
    }

    public Double getMaxLon() {
        return mMaxLon;
    }

    public static class Builder {
        private Double mMinLat;
        private Double mMinLon;
        private Double mMaxLat;
        private Double mMaxLon;

        public Double getMinLat() { return mMinLat; }

        public Double getMinLon() { return mMinLon; }

        public Double getMaxLat() { return mMaxLat; }

        public Double getMaxLon() { return mMaxLon; }

        public Builder setMinLat(Double minLat) {
            mMinLat = minLat;
            return this;
        }

        public Builder setMinLon(Double minLon) {
            mMinLon = minLon;
            return this;
        }

        public Builder setMaxLat(Double maxLat) {
            mMaxLat = maxLat;
            return this;
        }

        public Builder setMaxLon(Double maxLon) {
            mMaxLon = maxLon;
            return this;
        }

        public Bounds build() {
            return new Bounds(this);
        }
    }
}

File: parser/src/main/kotlin/io/ticofab/androidgpxparser/parser/domain/Bounds.kt
package io.ticofab.androidgpxparser.parser.domain

class Bounds private constructor(builder: Builder) {
    val minLat: Double?
    val minLon: Double?
    val maxLat: Double?
    val maxLon: Double?

    class Builder {
        var minLat: Double? = null
            private set
        var minLon: Double? = null
            private set
        var maxLat: Double? = null
            private set
        var maxLon: Double? = null
            private set

        fun setMinLat(minLat: Double?): Builder {
            this.minLat = minLat
            return this
        }

        fun setMinLon(minLon: Double?): Builder {
            this.minLon = minLon
            return this
        }

        fun setMaxLat(maxLat: Double?): Builder {
            this.maxLat = maxLat
            return this
        }

        fun setMaxLon(maxLon: Double?): Builder {
            this.maxLon = maxLon
            return this
        }

        fun build(): Bounds {
            return Bounds(this)
        }
    }

    init {
        minLat = builder.minLat
        minLon = builder.minLon
        maxLat = builder.maxLat
        maxLon = builder.maxLon
    }
}